src/client/datascience/commands/serverSelector.ts

```diff
@@ -8,20 +8,31 @@ import { ICommandManager } from '../../common/application/types';
 import { IDisposable } from '../../common/types';
 import { Commands } from '../constants';
 import { JupyterServerSelector } from '../jupyter/serverSelector';
+import { traceInfo } from '../../common/logger';
+import { Uri } from 'vscode'
 
 @injectable()
 export class JupyterServerSelectorCommand implements IDisposable {
     private readonly disposables: IDisposable[] = [];
     constructor(
         @inject(ICommandManager) private readonly commandManager: ICommandManager,
         @inject(JupyterServerSelector) private readonly serverSelector: JupyterServerSelector
-    ) {}
+    ) { }
     public register() {
         this.disposables.push(
             this.commandManager.registerCommand(
                 Commands.SelectJupyterURI,
-                (_, source: 'nativeNotebookStatusBar' | 'commandPalette' | 'toolbar' = 'commandPalette') =>
-                    this.serverSelector.selectJupyterURI(true, source),
+                (local: boolean = true, source: Uri | 'nativeNotebookStatusBar' | 'commandPalette' | 'toolbar' = 'commandPalette') => {
+
+                    if (source instanceof Uri) {
+                        traceInfo(`Setting Jupyter Server URI to remote: ${source}`);
+                        this.serverSelector.setJupyterURIToRemote(source.toString(true))
```

Comment:
I think both of these have to be marked with a 'void' on the front so the linter doesn't fail.